pkg/identity/cache/allocator_utils.go

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cache
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/pkg/allocator"
+	"github.com/cilium/cilium/pkg/identity"
+	identitymodel "github.com/cilium/cilium/pkg/identity/model"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/sirupsen/logrus"
+)
+
+type UpdateIDsFunc func(added, deleted IdentityCache)
+type ReleaseFunc func(ctx context.Context, id *identity.Identity, notifyOwner bool) (released bool, err error)
+type GetLocalIDsFunc func() map[identity.NumericIdentity]*identity.Identity
+
+var UnknownIdentity = identity.NewIdentity(identity.IdentityUnknown, labels.Labels{labels.IDNameUnknown: labels.NewLabel(labels.IDNameUnknown, "", labels.LabelSourceReserved)})
+
+func WithholdLocalIdentities(localIDs, localNodeIDs *LocalIdentityCache, nids []identity.NumericIdentity) {
```

Comment:
There seems to be several functions here that duplicate code from `allocator.go`, but I'm not sure that the code is reused. This can cause code divergence over time. Was that intentional?

Reply:
It wasn't intentionally. Thanks for catching it. I refactored it, and it's fixed.